main.go

```diff
@@ -274,3 +269,42 @@ func handlerGroupMessage(ctx context.Context, tgBot *bot.Bot, update *models.Upd
 		}
 	}
 }
+
+func isReply(update *models.Update) bool {
+	return update.Message.ReplyToMessage != nil && update.Message.ReplyToMessage != nil
+}
+
+// throwError is a function that returns an error with a specific error message
+// based on the errorID provided
+func throwError(errorID int, err error) error {
+	var errorMessage string
+	switch errorID {
+	case 1:
+		errorMessage = "Couldn't load .env file"
+	case 2:
+		errorMessage = "Couldn't load API_TOKEN variable from .env"
+	case 3:
+		errorMessage = "Couldn't construct Telegram bot object"
+	case 4:
+		errorMessage = "Couldn't reply to /help or /start command"
+	case 5:
+		errorMessage = "Failed to fetch cat"
+	case 6:
+		errorMessage = "Couldn't send cat pic"
+	case 7:
+		errorMessage = "You didn't type in a tag"
+	case 8:
+		errorMessage = "Failed to fetch cat by tag; status wasn't 200 OK"
+	case 9:
+		errorMessage = "SPECIFIED_CHAT_ID environment variable is not set"
+	case 10:
+		errorMessage = "Failed to send message to localhost"
+	case 11:
+		errorMessage = "Failed to decode response"
+	case 12:
+		errorMessage = "Failed to send warning message"
+	default:
+		errorMessage = "Unknown error occurred"
+	}
+	return fmt.Errorf("Error [%d]: %s; %w", errorID, errorMessage, err)
+}
```

Comment:
Hate to break it to you but this is just bad code.
Abstracting errors into a sort of enum only to then specify the error you're going to throw instead of inheriting it from the function where you're calling "throwError()" context's, only makes you have to jump to the function's definition whenever you check the code.
If you expect the function to break in a single way during a function, handle the error in there and if something unexpected happens just let Golang handle the error as it will log most information required to solve the issue.